app/src/api/controllers/userController.ts

```diff
@@ -0,0 +1,105 @@
+import { URL } from 'url';
+
+import express from 'express';
+
+import * as userService from '../services/user';
+import * as passwordResetService from '../services/passwordReset';
+import { TokenVerificationError } from '../services/passwordReset';
+
+function createUser(req, res) {
+    const user = req.body;
+    if (req.session.user_id) {
+        res.send({ error: 'Already signed in' });
+        return;
+    }
+
+    if (user.email) {
+        if (user.email != user.confirm_email) {
+            res.send({ error: 'Email did not match confirmation.' });
+            return;
+        }
+    } else {
+        user.email = null;
+    }
+
+    userService.createUser(user).then(function (result) {
+        if (result.user_id) {
+            req.session.user_id = result.user_id;
+            res.send({ user_id: result.user_id });
+        } else {
+            req.session.user_id = undefined;
+            res.send({ error: result.error });
+        }
+    }).catch(function (err) {
+        console.error(err);
+        res.send(err);
+    });
+}
+
+function getCurrentUser(req, res) {
+    if (!req.session.user_id) {
+        res.send({ error: 'Must be logged in' });
+        return;
+    }
+
+    userService.getUserById(req.session.user_id).then(function (user) {
+        res.send(user);
+    }).catch(function (error) {
+        res.send(error);
+    });
+}
+
+function deleteCurrentUser(req, res) {
+    if (!req.session.user_id) {
+        return res.send({ error: 'Must be logged in' });
+    }
+    console.log(`Deleting user ${req.session.user_id}`);
+
+    userService.deleteUser(req.session.user_id).then(
+        () => userService.logout(req.session)
+    ).then(() => {
+        res.send({ success: true });
+    }).catch(err => {
+        res.send({ error: err });
+    });
+}
+
+async function resetPassword(req: express.Request, res: express.Response) {
+    throw new Error('adsd');
```

Comment:
Always fails - for testing?

Reply:
ah, sorry about that... Yes indeed, not sure how I managed to leave it there. Pushing a fix now.